kpops/components/base_components/kubernetes_app.py

```diff
@@ -46,6 +45,9 @@ class KubernetesApp(PipelineComponent):
     :param schema_type: Used for schema generation, same as :param:`type`,
         defaults to "kubernetes-app"
     :type schema_type: Literal["kubernetes-app"], optional
+    :param validate_name: Whether to check if the name of the component is
+        compatible with Kubernetes, defaults to True
+    :type validate_name: bool, optional
```

Comment:
why is it optional? it even has a default value

Reply:
optional in the ReST means that you don't have to give it a value, not the same as `Optional[]`.

[reference](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format)

Reply:
ah ok, thanks for explaining. I find this very.. _unhelpful_ :D like I would expect the `type` attribute to only contain information about the allowed type and therefore understood optional as `... | None` 🤦🏻 
I guess most important is that we find a clear way that works for us

Reply:
Yeah, I agree that it is kinda counterintuitive. I am okay with abolishing the `optional` keyword in `:type ...:`, it is redundant when `:param ...:` already contains `, defaults to ...`